hcloud/_client.py

```diff
@@ -231,3 +233,44 @@ def request(  # type: ignore[no-untyped-def]
 
         # TODO: return an empty dict instead of an empty string when content == "".
         return content  # type: ignore[return-value]
+
+    def session(self, session: requests.Session) -> None:
+        """
+        Configure a custom :class:`Session <requests.Session>` to use when calling the API.
+
+        :param session: The session to use when making API requests.
+        """
+        self._requests_session = session
+
+    @contextmanager
+    def cached_session(self) -> Generator[Client, None, None]:
+        """
+        Provide a copy of the :class:`Client <hcloud.Client>` as context manager that
+        will cache all GET requests.
+
+        Cached response will not expire automatically, therefor the cached client must
```

Comment:
```suggestion
        Cached response will not expire automatically, therefore the cached client must
```

Perhaps you can even remove _automatically_ from the sentence, because there is no way to expire cached responses in any way, automatic or manual.